src/UI/Component/Table/Presentation.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/* Copyright (c) 2017 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Table;
+
+/**
+ * This describes a Presentation Table
+ */
+interface Presentation extends \ILIAS\UI\Component\Component {
+
+	/**
+	 * @return SignalGenerator
+	 */
+	public function getSignalGenerator();
+
+
+	/**
+	 * Get the title of the table.
+	 *
+	 * @return string
+	 */
+	public function getTitle();
+
+	/**
+	 * Get view controls to be shown in the header of the table.
+	 *
+	 * @return ILIAS\UI\Component\ViewControl[]
+	 */
+	public function getViewControls();
+
+	/**
+	 * Return the closure to map some data to the row.
+	 *
+	 * The closure MUST accept the following parameter
+	 * \PresentationRow $row
+	 * mixed $record
+	 * \Factory $ui_factory
+	 * array<string,mixed> $environment
+	 *
+	 * The closure MUST return \PresentationRow
+	 *
+	 * @return \Closure
+	 */
+	public function getRowMapping();
+
+	/**
+	 * Add a list of objects the mapping-closure needs for processing.
+	 *
+	 * @param array<string,mixed> 	$environment
```

Comment:
I think this could well be anything. It is just some thing that is passed to the `$row_mapping`, the row_mapping decides on the required structure.